### ISSUE TITLE
DSL generator generates incorrect code

### ISSUE THREAD

**Issue by asm0dey** [redundent/kotlin-xml-builder#55]
Hi there!

the generator generates code like 

```kotlin
open class Link(nodeName: String) : Node(nodeName) {
    init {
        xmlns = "http://www.w3.org/2005/Atom"
    }

    var `base`: String? by attributes
```
but `attributes` is a read-only map, so setters can't be delegated there.
One possible solution is to rewrite it like this:

```kotlin
    var base: String?
        get() = get("base")
        set(value) = set("base", value)
```
but it is also flawed since does not support attribute namespaces. Probably there should be a map storing attribute-namespace relation and then it could be used like

```kotlin
var base: String?
get() = get("base")
set(value) = set("base", value, nss["base"])
```

**Comment by redundent** [comment]
Hi,
I've been meaning to switch to using KotlinPoet for a while so thanks for getting that started. I'll take a look at your fork and see how it goes

**Comment by redundent** [comment]
This is proving to be more of a nightmare than I hoped. Turns out I never published the dsl builder library to Maven Central after jcenter went offline so I'm having to set all that up again.

**Comment by asm0dey** [comment]
Do you need help? I have experience with it, can help with GH Actions, for example, you'll only need to put your gpg private key, password and id into GH Secrets

**Comment by redundent** [comment]
No, I can handle it. Thanks for the offer though. I just need to find the time to deal with getting it all sorted. I'm hoping to have some time this weekend to get it going.

**Comment by asm0dey** [comment]
If anything, here is the example with all the modern best practices: https://github.com/asm0dey/dummylib-multiplatform/blob/main/convention-plugins/src/main/kotlin/module.publication.gradle.kts

Here it's used for a KMP project, but the same approach can be used for JVM.

And here is the deployment action: https://github.com/asm0dey/dummylib-multiplatform/blob/main/.github/workflows/deploy.yml

**Comment by redundent** [comment]
I just published v1.9.0 of the kotlin-xml-dsl-generator which contains the fix for incorrect code generation. I have a ton of stuff to clean up in the repo and I didn't touch the kotlinpoet changes yet. I will do that soon.

**Comment by asm0dey** [comment]
Awesome!
If you want, I can submit a PR, just briefly describe your desired result

**Comment by redundent** [comment]
Thanks but I'll take care of it. While fixing the attribute issue, I noticed how much of an embarrassing mess that code was an now I just want to set it on fire and rewrite it all.
